corehq/apps/hqadmin/views.py

```diff
@@ -490,11 +492,24 @@ def get_context_data(self, **kwargs):
         context = super(AdminRestoreView, self).get_context_data(**kwargs)
         response, timing_context = self._get_restore_response()
         timing_context = timing_context or TimingContext(self.user.username)
-        string_payload = ''.join(response.streaming_content)
-        xml_payload = etree.fromstring(string_payload)
-        restore_id_element = xml_payload.find('{{{0}}}Sync/{{{0}}}restore_id'.format(SYNC_XMLNS))
-        num_cases = len(xml_payload.findall('{http://commcarehq.org/case/transaction/v2}case'))
-        formatted_payload = etree.tostring(xml_payload, pretty_print=True)
+        if isinstance(response, StreamingHttpResponse):
+            string_payload = ''.join(response.streaming_content)
+            xml_payload = etree.fromstring(string_payload)
+            restore_id_element = xml_payload.find('{{{0}}}Sync/{{{0}}}restore_id'.format(SYNC_XMLNS))
+            num_cases = len(xml_payload.findall('{http://commcarehq.org/case/transaction/v2}case'))
+            formatted_payload = etree.tostring(xml_payload, pretty_print=True)
+        else:
+            # get_restore_response return HttpResponse 401 (permissions), 404 (user not found) or 412
+            if response.status_code in (401, 404):
```

Comment:
should `412` be in this list too based on above comment? that makes sense to me as those errors are also useful.